src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/QueryablesBuilder.java

```diff
@@ -55,15 +56,21 @@ public QueryablesBuilder forType(FeatureTypeInfo ft) throws IOException {
         this.queryables.setCollectionId(ft.prefixedName());
         this.queryables.setTitle(Optional.ofNullable(ft.getTitle()).orElse(ft.prefixedName()));
         this.queryables.setDescription(ft.getDescription());
-        return forType((SimpleFeatureType) ft.getFeatureType());
+        return forType(ft.getFeatureType());
     }
 
-    public QueryablesBuilder forType(SimpleFeatureType ft) {
+    public QueryablesBuilder forType(FeatureType ft) {
         Map<String, Schema> properties =
-                ft.getAttributeDescriptors().stream()
+                ft.getDescriptors().stream()
+                        .filter(
+                                ad -> // ignore feature chaining links, they might be duplicated
+                                !ad.getName()
+                                                .equals(
+                                                        ComplexFeatureConstants
+                                                                .FEATURE_CHAINING_LINK_NAME))
                         .collect(
                                 Collectors.toMap(
-                                        ad -> ad.getLocalName(),
+                                        ad -> ad.getName().getLocalPart(),
```

Comment:
Hum... so the queryables are the top-level attributes?
What if one of the attributes is complex though, like holding a feature or a collection? It would not be usable in CQL2.

Reply:
Admitted, this is merely a fix to get rid of the ugly stack trace, not proper complex feature functionality. The problem however is that it is not possible to provide an exhaustive list of possible xpaths for complex features, because there may be infinite recursion (and there usually is)

Reply:
Makes sense, it's a community module after all. I did not mean to have full queriables, more like skipping attributes that cannot work because of their complex nature.